From 42afc1f652d28dce3c5b9915d0cd3087a144e445 Mon Sep 17 00:00:00 2001
From: Alejandro del Castillo <alejandro.delcastillo@ni.com>
Date: Mon, 18 Feb 2019 11:25:43 -0600
Subject: [PATCH] opkg-build: do not set mtime on data.tar.X

Opkg should make sure the binaries it creates are created in a
repoducibe way, but it shouldn't modify the mtime of the binaries it
consumes. Doing so has side effects, for exmaple, Python pyc files are
invalidated.

Signed-off-by: Alejandro del Castillo <alejandro.delcastillo@ni.com>

Upstream-Status: Submitted
---
 opkg-build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/opkg-build b/opkg-build
index 13782ee..363cd01 100755
--- a/opkg-build
+++ b/opkg-build
@@ -301,7 +301,7 @@ build_date="${SOURCE_DATE_EPOCH:-$(date +%s)}"
 
 ( cd $pkg_dir/$CONTROL && find . -type f > $tmp_dir/control_list )
 ( cd $pkg_dir && find . -path ./$CONTROL -prune -o -print > $tmp_dir/file_list )
-( cd $pkg_dir && tar $ogargs $tsortargs --no-recursion --mtime=@$build_date -c $tarformat -T $tmp_dir/file_list | $compressor $compressorargs > $tmp_dir/data.tar.$cext )
+( cd $pkg_dir && tar $ogargs $tsortargs --no-recursion -c $tarformat -T $tmp_dir/file_list | $compressor $compressorargs > $tmp_dir/data.tar.$cext )
 ( cd $pkg_dir/$CONTROL && tar $ogargs $tsortargs --no-recursion --mtime=@$build_date -c $tarformat -T $tmp_dir/control_list | gzip $zipargs > $tmp_dir/control.tar.gz )
 rm $tmp_dir/file_list
 rm $tmp_dir/control_list
-- 
2.20.1

